log/internal_logger.go

```diff
@@ -0,0 +1,54 @@
+package log
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"net/http"
+)
+
+var analyticsServerURL = "https://bitrise-step-analytics.herokuapp.com"
+
+// Message represents a line in a log
+type Message struct{
+	LogLevel string `json:"log_level"`
+	Message string `json:"message"`
+	Data map[string]interface{} `json:"data"`
+}
+
+func init() {
+	if url := os.Getenv("ANALYTICS_SERVER_URL"); url != "" {
+		analyticsServerURL = url
+	}
+}
+
+// SendToInternal sends the log message to the configured analytics server
+func (lm Message) SendToInternal(stepID, tag string, data map[string]interface{}) {
+	if lm.LogLevel == "" {
```

Comment:
In cases we call this method we want something to be stored anyway, I think the network call should happen even if this information is missing.